src/app/modules/core/services/beacon-node.service.ts

```diff
@@ -25,16 +36,41 @@ export class BeaconNodeService {
   ) { }
 
   private apiUrl = this.environmenter.env.validatorEndpoint;
-  private beaconNodeEndpoint: string;
-
-  // Observables.
-  status$ = this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).pipe(
-    tap((res: NodeConnectionResponse) => {
-      this.beaconNodeEndpoint = res.beaconNodeEndpoint + BEACON_API_SUFFIX;
-    }),
-  );
+  beaconState: BeaconState = new BeaconState();
+
+  private get getBeaconEndpoint$(): BehaviorSubject<string> {
+    if (!this.beaconState.beaconNodeEndpoint$.getValue()) {
+      this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).subscribe(
+          (res: NodeConnectionResponse) => {
+            this.beaconState.beaconNodeEndpoint$.next("http://" + res.beaconNodeEndpoint + BEACON_API_SUFFIX);
+          },
+          (error) => {
+          }
+        );
+    }
+    return this.beaconState.beaconNodeEndpoint$;
+  }
+
+  private updateBalances(): void {
+    this.getBeaconEndpoint$.pipe(
+      filter((beacanNodeEnpoint) => {
+        return beacanNodeEnpoint !== undefined;
+      }),
+      map(beacanNodeEnpoint => {
+        return this.http.get<ValidatorBalances>(`${beacanNodeEnpoint}/validator/balances`)
+          .subscribe((result) => {
+            this.beaconState.balances$.next(result)
+            this.beaconState.connected$.next(true);
+          },
+          (err) => {
+            this.beaconState.connected$.next(false);
+          });
+      })
+    ).subscribe();
+  }
+
   statusPoll$ = interval(POLLING_INTERVAL).pipe(
     startWith(0),
-    mergeMap(_ => this.status$),
-  );
+    tap(() => this.updateBalances()),
```

Comment:
Let's avoid side-effects like these

Reply:
you mean the , at the end of "tap(() => this.updateBalances()),"?